# **Computational Drug Discovery Project: Data Preprocessing**
By Mathew Kuruvilla

Based on the Drug Discovery Project taught by Chanin Nantasenamat [*'Data Professor' YouTube channel*](http://youtube.com/dataprofessor)

In this project, I will be building a machine learning model using bioactivity data from ChEMBL for coronavirus replicase polyprotein 1ab inhibitors.
This Jupyter notebook will focus on data preprocessing steps and compile bioactivity class data.

---

## **Installing libraries**

In [1]:
! pip install chembl_webresource_client

  Using cached chembl_webresource_client-0.10.9-py3-none-any.whl.metadata (1.4 kB)
  Using cached requests_cache-1.2.1-py3-none-any.whl.metadata (9.9 kB)
  Using cached easydict-1.13-py3-none-any.whl.metadata (4.2 kB)
  Using cached cattrs-24.1.3-py3-none-any.whl.metadata (8.4 kB)
Using cached chembl_webresource_client-0.10.9-py3-none-any.whl (55 kB)
Using cached requests_cache-1.2.1-py3-none-any.whl (61 kB)
Using cached easydict-1.13-py3-none-any.whl (6.8 kB)
Using cached cattrs-24.1.3-py3-none-any.whl (66 kB)


## **Importing libraries**

In [2]:
import pandas as pd
from chembl_webresource_client.new_client import new_client

## **Search for Target protein**

### **Target search for coronavirus**

In [3]:
# Target search for coronavirus
target = new_client.target
target_query = target.search('coronavirus')
targets = pd.DataFrame.from_dict(target_query)
targets

cross_references                                           organism  \
0               []                                        Coronavirus   
1               []                                 Feline coronavirus   
2               []                                 Murine coronavirus   
3               []                                 Canine coronavirus   
4               []                             Human coronavirus 229E   
5               []                             Human coronavirus OC43   
6               []  Severe acute respiratory syndrome-related coro...   
7               []  Middle East respiratory syndrome-related coron...   
8               []  Severe acute respiratory syndrome-related coro...   
9               []    Severe acute respiratory syndrome coronavirus 2   

                                           pref_name  score  \
0                                        Coronavirus   17.0   
1                                 Feline coronavirus   14.0   
2                                 Murine coronavirus   14.0   
3                                 Canine coronavirus   14.0   
4                             Human coronavirus 229E   13.0   
5                             Human coronavirus OC43   13.0   
6                SARS coronavirus 3C-like proteinase   10.0   
7  Middle East respiratory syndrome-related coron...    9.0   
8                          Replicase polyprotein 1ab    4.0   
9                          Replicase polyprotein 1ab    4.0   

   species_group_flag target_chembl_id  \
0               False     CHEMBL613732   
1               False     CHEMBL612744   
2               False    CHEMBL5209664   
3               False    CHEMBL5291668   
4               False     CHEMBL613837   
5               False    CHEMBL5209665   
6               False       CHEMBL3927   
7               False    CHEMBL4296578   
8               False       CHEMBL5118   
9               False    CHEMBL4523582   

                                   target_components     target_type   tax_id  
0                                                 []        ORGANISM    11119  
1                                                 []        ORGANISM    12663  
2                                                 []        ORGANISM   694005  
3                                                 []        ORGANISM    11153  
4                                                 []        ORGANISM    11137  
5                                                 []        ORGANISM    31631  
6  [{'accession': 'P0C6U8', 'component_descriptio...  SINGLE PROTEIN   694009  
7                                                 []        ORGANISM  1335626  
8  [{'accession': 'P0C6X7', 'component_descriptio...  SINGLE PROTEIN   694009  
9  [{'accession': 'P0DTD1', 'component_descriptio...  SINGLE PROTEIN  2697049

### **Select and retrieve bioactivity data for *replicase polyprotein 1ab* (tenth entry)**

In [4]:
selected_target = targets.target_chembl_id[9]
selected_target

'CHEMBL4523582'

Here, we will retrieve only bioactivity data for *Replicase polyprotein 1ab* (CHEMBL4523582) that are reported as IC$_{50}$. <br>
IC$_{50}$ is a measure of how much compound is needed to inhibit the enzyme by 50% recorded as values in nM (nanomolar) unit. We are filtering specifically for experiments that tested measured how much of a drug is needed to inhibit 50% of the protease’s activity. <br>
The replicase polyprotein 1ab was chosen for this study because it is a huge polyprotein that coronaviruses make inside infected cells. Its main job is to copy (replicate) and control the virus’s RNA genome. Inhibitors targeting the replicase polyprotein 1ab would block the virus’s ability to copy its RNA, stopping it from making new viruses inside the host. Without a working replication complex, the infection can't spread and the virus eventually dies off.

In [5]:
activity = new_client.activity
res = activity.filter(target_chembl_id=selected_target).filter(type="IC50")
res

[{'action_type': None, 'activity_comment': 'Dtt Insensitive', 'activity_id': 19964199, 'activity_properties': [], 'assay_chembl_id': 'CHEMBL4495583', 'assay_description': 'SARS-CoV-2 3CL-Pro protease inhibition IC50 determined by FRET kind of response from peptide substrate', 'assay_type': 'F', 'assay_variant_accession': None, 'assay_variant_mutation': None, 'bao_endpoint': 'BAO_0000190', 'bao_format': 'BAO_0000019', 'bao_label': 'assay format', 'canonical_smiles': 'Cc1c(OCC(F)(F)F)ccnc1C[S+]([O-])c1nc2ccccc2[nH]1', 'data_validity_comment': None, 'data_validity_description': None, 'document_chembl_id': 'CHEMBL4495564', 'document_journal': None, 'document_year': 2020, 'ligand_efficiency': None, 'molecule_chembl_id': 'CHEMBL480', 'molecule_pref_name': 'LANSOPRAZOLE', 'parent_molecule_chembl_id': 'CHEMBL480', 'pchembl_value': '6.41', 'potential_duplicate': 0, 'qudt_units': 'http://www.openphacts.org/units/Nanomolar', 'record_id': 3341963, 'relation': '=', 'src_id': 52, 'standard_flag': 1,

In [6]:
df = pd.DataFrame.from_dict(res)
df.shape

(3660, 46)

In [7]:
with pd.option_context('display.max_columns', None): 
    display(df.head())

action_type activity_comment  activity_id activity_properties  \
0        None  Dtt Insensitive     19964199                  []   
1        None  Dtt Insensitive     19964200                  []   
2        None  Dtt Insensitive     19964201                  []   
3        None  Dtt Insensitive     19964202                  []   
4        None  Dtt Insensitive     19964203                  []   

  assay_chembl_id                                  assay_description  \
0   CHEMBL4495583  SARS-CoV-2 3CL-Pro protease inhibition IC50 de...   
1   CHEMBL4495583  SARS-CoV-2 3CL-Pro protease inhibition IC50 de...   
2   CHEMBL4495583  SARS-CoV-2 3CL-Pro protease inhibition IC50 de...   
3   CHEMBL4495583  SARS-CoV-2 3CL-Pro protease inhibition IC50 de...   
4   CHEMBL4495583  SARS-CoV-2 3CL-Pro protease inhibition IC50 de...   

  assay_type assay_variant_accession assay_variant_mutation bao_endpoint  \
0          F                    None                   None  BAO_0000190   
1          F                    None                   None  BAO_0000190   
2          F                    None                   None  BAO_0000190   
3          F                    None                   None  BAO_0000190   
4          F                    None                   None  BAO_0000190   

    bao_format     bao_label  \
0  BAO_0000019  assay format   
1  BAO_0000019  assay format   
2  BAO_0000019  assay format   
3  BAO_0000019  assay format   
4  BAO_0000019  assay format   

                                    canonical_smiles data_validity_comment  \
0   Cc1c(OCC(F)(F)F)ccnc1C[S+]([O-])c1nc2ccccc2[nH]1                  None   
1                              Cc1c(-c2cnccn2)ssc1=S                  None   
2            O=c1sn(-c2cccc3ccccc23)c(=O)n1Cc1ccccc1                  None   
3  O=C(O[C@@H]1Cc2c(O)cc(O)cc2O[C@@H]1c1cc(O)c(O)...                  None   
4                            O=C1C=Cc2cc(Br)ccc2C1=O                  None   

  data_validity_description document_chembl_id document_journal  \
0                      None      CHEMBL4495564             None   
1                      None      CHEMBL4495564             None   
2                      None      CHEMBL4495564             None   
3                      None      CHEMBL4495564             None   
4                      None      CHEMBL4495564             None   

   document_year ligand_efficiency molecule_chembl_id  \
0           2020              None          CHEMBL480   
1           2020              None       CHEMBL178459   
2           2020              None      CHEMBL3545157   
3           2020              None       CHEMBL297453   
4           2020              None      CHEMBL4303595   

        molecule_pref_name parent_molecule_chembl_id pchembl_value  \
0             LANSOPRAZOLE                 CHEMBL480          6.41   
1                 OLTIPRAZ              CHEMBL178459          6.68   
2               TIDEGLUSIB             CHEMBL3545157          7.10   
3  EPIGALOCATECHIN GALLATE              CHEMBL297453          5.80   
4                     None             CHEMBL4303595          7.40   

   potential_duplicate                                 qudt_units  record_id  \
0                    0  http://www.openphacts.org/units/Nanomolar    3341963   
1                    0  http://www.openphacts.org/units/Nanomolar    3341991   
2                    0  http://www.openphacts.org/units/Nanomolar    3342067   
3                    0  http://www.openphacts.org/units/Nanomolar    3342156   
4                    0  http://www.openphacts.org/units/Nanomolar    3342307   

  relation  src_id  standard_flag standard_relation standard_text_value  \
0        =      52              1                 =                None   
1        =      52              1                 =                None   
2        =      52              1                 =                None   
3        =      52              1                 =                None   
4        =      52      

The **standard_value** column tells us how much of the drug is needed to reach the IC$_{50}$. The lower the number, the more effective the particular drug is at inhibiting the activity of the protease.

In [8]:
df.standard_type.unique()

array(['IC50'], dtype=object)

Finally we will save the resulting bioactivity data to a CSV file **bioactivity_data.csv**.

In [20]:
df.to_csv('bioactivity_data.csv', index=False)

## **Handling missing data**
If any compound has missing values for the **standard_value** column drop it.

In [9]:
df2_1 = df[df.standard_value.notna()].reset_index()
dropped_rows_sv = df[df.standard_value.isna()]
print(len(dropped_rows_sv))
print(len(df2_1))
df2_1

104
3556


index action_type activity_comment  activity_id  \
0         0        None  Dtt Insensitive     19964199   
1         1        None  Dtt Insensitive     19964200   
2         2        None  Dtt Insensitive     19964201   
3         3        None  Dtt Insensitive     19964202   
4         4        None  Dtt Insensitive     19964203   
...     ...         ...              ...          ...   
3551   3655        None             None     25739541   
3552   3656        None             None     25739542   
3553   3657        None             None     25739543   
3554   3658        None             None     25739544   
3555   3659        None             None     25739545   

                                    activity_properties assay_chembl_id  \
0                                                    []   CHEMBL4495583   
1                                                    []   CHEMBL4495583   
2                                                    []   CHEMBL4495583   
3                                                    []   CHEMBL4495583   
4                                                    []   CHEMBL4495583   
...                                                 ...             ...   
3551  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3552  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3553  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3554  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3555  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   

                                      assay_description assay_type  \
0     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
1     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
2     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
3     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
4     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
...                                                 ...        ...   
3551  Functional biochemical assay to identify treat...          F   
3552  Functional biochemical assay to identify treat...          F   
3553  Functional biochemical assay to identify treat...          F   
3554  Functional biochemical assay to identify treat...          F   
3555  Functional biochemical assay to identify treat...          F   

     assay_variant_accession assay_variant_mutation  ...  \
0                       None                   None  ...   
1                       None                   None  ...   
2                       None                   None  ...   
3                       None                   None  ...   
4                       None                   None  ...   
...                      ...                    ...  ...   
3551                    None                   None  ...   
3552                    None                   None  ...   
3553                    None                   None  ...   
3554                    None                   None  ...   
3555                    None                   None  ...   

                                      target_organism  \
0     Severe acute respiratory syndrome coronavirus 2   
1     Severe acute respiratory syndrome coronavirus 2   
2     Severe acute respiratory syndrome coronavirus 2   
3     Severe acute respiratory syndrome coronavirus 2   
4     Severe acute respiratory syndrome coronavirus 2   
...                                               ...   
3551  Severe acute respiratory syndrome coronavirus 2   
3552  Severe acute respiratory syndrome coronavirus 2   
3553  Severe acute respiratory syndrome coronavirus 2   
3554  Severe acute respiratory syndrome coronavirus 2   
3555  Severe acute respiratory syndrome coronavirus 2   

               target_pref_name target_tax_id text_value  toid  type units  \
0     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
1     Replicase p

In [10]:
df2_2 = df2_1[df2_1.canonical_smiles.notna()].reset_index()
dropped_rows_cs = df2_1[df2_1.canonical_smiles.isna()]
print(len(dropped_rows_cs))
print(len(df2_2))
df2_2

10
3546


level_0  index action_type activity_comment  activity_id  \
0           0      0        None  Dtt Insensitive     19964199   
1           1      1        None  Dtt Insensitive     19964200   
2           2      2        None  Dtt Insensitive     19964201   
3           3      3        None  Dtt Insensitive     19964202   
4           4      4        None  Dtt Insensitive     19964203   
...       ...    ...         ...              ...          ...   
3541     3551   3655        None             None     25739541   
3542     3552   3656        None             None     25739542   
3543     3553   3657        None             None     25739543   
3544     3554   3658        None             None     25739544   
3545     3555   3659        None             None     25739545   

                                    activity_properties assay_chembl_id  \
0                                                    []   CHEMBL4495583   
1                                                    []   CHEMBL4495583   
2                                                    []   CHEMBL4495583   
3                                                    []   CHEMBL4495583   
4                                                    []   CHEMBL4495583   
...                                                 ...             ...   
3541  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3542  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3543  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3544  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
3545  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   

                                      assay_description assay_type  \
0     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
1     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
2     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
3     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
4     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
...                                                 ...        ...   
3541  Functional biochemical assay to identify treat...          F   
3542  Functional biochemical assay to identify treat...          F   
3543  Functional biochemical assay to identify treat...          F   
3544  Functional biochemical assay to identify treat...          F   
3545  Functional biochemical assay to identify treat...          F   

     assay_variant_accession  ...  \
0                       None  ...   
1                       None  ...   
2                       None  ...   
3                       None  ...   
4                       None  ...   
...                      ...  ...   
3541                    None  ...   
3542                    None  ...   
3543                    None  ...   
3544                    None  ...   
3545                    None  ...   

                                      target_organism  \
0     Severe acute respiratory syndrome coronavirus 2   
1     Severe acute respiratory syndrome coronavirus 2   
2     Severe acute respiratory syndrome coronavirus 2   
3     Severe acute respiratory syndrome coronavirus 2   
4     Severe acute respiratory syndrome coronavirus 2   
...                                               ...   
3541  Severe acute respiratory syndrome coronavirus 2   
3542  Severe acute respiratory syndrome coronavirus 2   
3543  Severe acute respiratory syndrome coronavirus 2   
3544  Severe acute respiratory syndrome coronavirus 2   
3545  Severe acute respiratory syndrome coronavirus 2   

               target_pref_name target_tax_id text_value  toid  type units  \
0     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
1     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
2     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
3     Replicase polyprotein 1

In [11]:
df2_3 = df2_2.drop_duplicates(subset='canonical_smiles').reset_index(drop=True)
dropped_rows_dupe = len(df2_2) - len(df2_3)
print(dropped_rows_dupe)
print(len(df2_3))
df2_3

1145
2401


level_0  index action_type               activity_comment  activity_id  \
0           0      0        None                Dtt Insensitive     19964199   
1           1      1        None                Dtt Insensitive     19964200   
2           2      2        None                Dtt Insensitive     19964201   
3           3      3        None                Dtt Insensitive     19964202   
4           4      4        None                Dtt Insensitive     19964203   
...       ...    ...         ...                            ...          ...   
2396     3529   3633        None  Falls Outside The Dose Series     25739519   
2397     3530   3634        None                           None     25739520   
2398     3538   3642        None  Falls Outside The Dose Series     25739528   
2399     3541   3645        None                           None     25739531   
2400     3542   3646        None                           None     25739532   

                                    activity_properties assay_chembl_id  \
0                                                    []   CHEMBL4495583   
1                                                    []   CHEMBL4495583   
2                                                    []   CHEMBL4495583   
3                                                    []   CHEMBL4495583   
4                                                    []   CHEMBL4495583   
...                                                 ...             ...   
2396  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2397  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2398  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2399  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2400  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   

                                      assay_description assay_type  \
0     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
1     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
2     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
3     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
4     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
...                                                 ...        ...   
2396  Functional biochemical assay to identify treat...          F   
2397  Functional biochemical assay to identify treat...          F   
2398  Functional biochemical assay to identify treat...          F   
2399  Functional biochemical assay to identify treat...          F   
2400  Functional biochemical assay to identify treat...          F   

     assay_variant_accession  ...  \
0                       None  ...   
1                       None  ...   
2                       None  ...   
3                       None  ...   
4                       None  ...   
...                      ...  ...   
2396                    None  ...   
2397                    None  ...   
2398                    None  ...   
2399                    None  ...   
2400                    None  ...   

                                      target_organism  \
0     Severe acute respiratory syndrome coronavirus 2   
1     Severe acute respiratory syndrome coronavirus 2   
2     Severe acute respiratory syndrome coronavirus 2   
3     Severe acute respiratory syndrome coronavirus 2   
4     Severe acute respiratory syndrome coronavirus 2   
...                                               ...   
2396  Severe acute respiratory syndrome coronavirus 2   
2397  Severe acute respiratory syndrome coronavirus 2   
2398  Severe acute respiratory syndrome coronavirus 2   
2399  Severe acute respiratory syndrome coronavirus 2   
2400  Severe acute respiratory syndrome coronavirus 2   

               target_pref_name target_tax_id text_value  toid  type units  \
0     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
1     Replicase p

In [12]:
df3 = df2_3[df2_3.standard_type == 'IC50'].reset_index(drop=True)
dropped_rows_ic50 = len(df2_3) - len(df3)
print(dropped_rows_ic50)
print(len(df3))
df3

0
2401


level_0  index action_type               activity_comment  activity_id  \
0           0      0        None                Dtt Insensitive     19964199   
1           1      1        None                Dtt Insensitive     19964200   
2           2      2        None                Dtt Insensitive     19964201   
3           3      3        None                Dtt Insensitive     19964202   
4           4      4        None                Dtt Insensitive     19964203   
...       ...    ...         ...                            ...          ...   
2396     3529   3633        None  Falls Outside The Dose Series     25739519   
2397     3530   3634        None                           None     25739520   
2398     3538   3642        None  Falls Outside The Dose Series     25739528   
2399     3541   3645        None                           None     25739531   
2400     3542   3646        None                           None     25739532   

                                    activity_properties assay_chembl_id  \
0                                                    []   CHEMBL4495583   
1                                                    []   CHEMBL4495583   
2                                                    []   CHEMBL4495583   
3                                                    []   CHEMBL4495583   
4                                                    []   CHEMBL4495583   
...                                                 ...             ...   
2396  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2397  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2398  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2399  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   
2400  [{'comments': None, 'relation': '=', 'result_f...   CHEMBL5441457   

                                      assay_description assay_type  \
0     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
1     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
2     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
3     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
4     SARS-CoV-2 3CL-Pro protease inhibition IC50 de...          F   
...                                                 ...        ...   
2396  Functional biochemical assay to identify treat...          F   
2397  Functional biochemical assay to identify treat...          F   
2398  Functional biochemical assay to identify treat...          F   
2399  Functional biochemical assay to identify treat...          F   
2400  Functional biochemical assay to identify treat...          F   

     assay_variant_accession  ...  \
0                       None  ...   
1                       None  ...   
2                       None  ...   
3                       None  ...   
4                       None  ...   
...                      ...  ...   
2396                    None  ...   
2397                    None  ...   
2398                    None  ...   
2399                    None  ...   
2400                    None  ...   

                                      target_organism  \
0     Severe acute respiratory syndrome coronavirus 2   
1     Severe acute respiratory syndrome coronavirus 2   
2     Severe acute respiratory syndrome coronavirus 2   
3     Severe acute respiratory syndrome coronavirus 2   
4     Severe acute respiratory syndrome coronavirus 2   
...                                               ...   
2396  Severe acute respiratory syndrome coronavirus 2   
2397  Severe acute respiratory syndrome coronavirus 2   
2398  Severe acute respiratory syndrome coronavirus 2   
2399  Severe acute respiratory syndrome coronavirus 2   
2400  Severe acute respiratory syndrome coronavirus 2   

               target_pref_name target_tax_id text_value  toid  type units  \
0     Replicase polyprotein 1ab       2697049       None  None  IC50    uM   
1     Replicase p

For this dataset there were 104 rows with missing standard_value data, 10 missing canonical_smiles data, 1145 duplicate canonical_smiles data, and 0 non IC50 measurements.

## **Data pre-processing of the bioactivity data**

### **Labeling compounds as either being active, inactive or intermediate**
The bioactivity data is in the IC$_{50}$ unit. Compounds having values of less than 1,000 nM will be considered to be **active** while those greater than 10,000 nM will be considered to be **inactive**. As for those values in between 1,000 and 10,000 nM will be referred to as **intermediate**. 

In [13]:
bioactivity_class = []
for i in df3.standard_value:
  if float(i) >= 10000:
    bioactivity_class.append("inactive")
  elif float(i) <= 1000:
    bioactivity_class.append("active")
  else:
    bioactivity_class.append("intermediate")

bc_set = set(bioactivity_class)
print(bc_set)

{'intermediate', 'active', 'inactive'}


### **Compiling bioactivity data into one dataframe**

In [14]:
selection = ['molecule_chembl_id', 'canonical_smiles', 'standard_value']
df4 = df3[selection]
df4

molecule_chembl_id                                   canonical_smiles  \
0             CHEMBL480   Cc1c(OCC(F)(F)F)ccnc1C[S+]([O-])c1nc2ccccc2[nH]1   
1          CHEMBL178459                              Cc1c(-c2cnccn2)ssc1=S   
2         CHEMBL3545157            O=c1sn(-c2cccc3ccccc23)c(=O)n1Cc1ccccc1   
3          CHEMBL297453  O=C(O[C@@H]1Cc2c(O)cc(O)cc2O[C@@H]1c1cc(O)c(O)...   
4         CHEMBL4303595                            O=C1C=Cc2cc(Br)ccc2C1=O   
...                 ...                                                ...   
2396      CHEMBL5441534  CC(C)[C@H](Nc1ncncc1-c1nc[nH]n1)c1ccc2c(c1)S(=...   
2397      CHEMBL5441398  COc1ccc([C@H](Cc2nnn[nH]2)NC(=O)c2ncnc3[nH]ccc...   
2398      CHEMBL5442105     CC(C)[C@@H](Oc1ncnc2[nH]ccc12)c1ccc2c(c1)OCCO2   
2399      CHEMBL5441739  CC(C)[C@@H](Nc1ncnc2[nH]c(Br)cc12)c1ccc2c(c1)S...   
2400      CHEMBL5441567  CC(C)[C@H](Nc1ncnc2c1CCN2)c1ccc2c(c1)S(=O)(=O)...   

     standard_value  
0             390.0  
1             210.0  
2              80.0  
3            1580.0  
4              40.0  
...             ...  
2396        99500.0  
2397          349.0  
2398        99500.0  
2399          336.0  
2400        52128.0  

[2401 rows x 3 columns]

In [15]:
bc_pd = pd.DataFrame(bioactivity_class)
bc_pd.columns= ['bioactivity_class']
bc_pd

bioactivity_class
0               active
1               active
2               active
3         intermediate
4               active
...                ...
2396          inactive
2397            active
2398          inactive
2399            active
2400          inactive

[2401 rows x 1 columns]

In [16]:
df5 = pd.concat([df4,bc_pd], axis=1)
df5

molecule_chembl_id                                   canonical_smiles  \
0             CHEMBL480   Cc1c(OCC(F)(F)F)ccnc1C[S+]([O-])c1nc2ccccc2[nH]1   
1          CHEMBL178459                              Cc1c(-c2cnccn2)ssc1=S   
2         CHEMBL3545157            O=c1sn(-c2cccc3ccccc23)c(=O)n1Cc1ccccc1   
3          CHEMBL297453  O=C(O[C@@H]1Cc2c(O)cc(O)cc2O[C@@H]1c1cc(O)c(O)...   
4         CHEMBL4303595                            O=C1C=Cc2cc(Br)ccc2C1=O   
...                 ...                                                ...   
2396      CHEMBL5441534  CC(C)[C@H](Nc1ncncc1-c1nc[nH]n1)c1ccc2c(c1)S(=...   
2397      CHEMBL5441398  COc1ccc([C@H](Cc2nnn[nH]2)NC(=O)c2ncnc3[nH]ccc...   
2398      CHEMBL5442105     CC(C)[C@@H](Oc1ncnc2[nH]ccc12)c1ccc2c(c1)OCCO2   
2399      CHEMBL5441739  CC(C)[C@@H](Nc1ncnc2[nH]c(Br)cc12)c1ccc2c(c1)S...   
2400      CHEMBL5441567  CC(C)[C@H](Nc1ncnc2c1CCN2)c1ccc2c(c1)S(=O)(=O)...   

     standard_value bioactivity_class  
0             390.0            active  
1             210.0            active  
2              80.0            active  
3            1580.0      intermediate  
4              40.0            active  
...             ...               ...  
2396        99500.0          inactive  
2397          349.0            active  
2398        99500.0          inactive  
2399          336.0            active  
2400        52128.0          inactive  

[2401 rows x 4 columns]

Saves dataframe to CSV file

In [17]:
df5.to_csv('bioactivity_preprocessed_data.csv', index=False)